public/app/core/components/AppChrome/ExtensionSidebar/ExtensionToolbarItem.tsx

```diff
@@ -0,0 +1,67 @@
+import { useState } from 'react';
+
+import { Dropdown, Menu, ToolbarButton } from '@grafana/ui';
+
+import { getComponentIdFromComponentMeta, useExtensionSidebarContext } from './ExtensionSidebarProvider';
+
+export function ExtensionToolbarItem() {
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const { availableComponents, dockedComponentId, setDockedComponentId, isOpen } = useExtensionSidebarContext();
+
+  if (availableComponents.size === 0) {
+    return null;
+  }
+
+  // get a flat list of all components with their pluginId
+  const components = Array.from(availableComponents.entries()).flatMap(([pluginId, { exposedComponents }]) =>
+    exposedComponents.map((c) => ({ ...c, pluginId }))
+  );
+
+  if (components.length === 0) {
+    return null;
+  }
+
+  if (components.length === 1) {
+    return (
+      <ToolbarButton
+        icon="web-section"
+        variant={isOpen ? 'active' : 'default'}
+        tooltip={components[0].description}
+        onClick={() => {
+          if (isOpen) {
+            setDockedComponentId(undefined);
+          } else {
+            setDockedComponentId(getComponentIdFromComponentMeta(components[0].pluginId, components[0]));
+          }
+        }}
+      />
+    );
+  }
+
+  const MenuItems = (
+    <Menu>
+      {components.map((c) => {
+        const id = getComponentIdFromComponentMeta(c.pluginId, c);
+        return (
+          <Menu.Item
+            key={id}
+            active={dockedComponentId === id}
+            label={c.title}
+            onClick={() => {
+              if (isOpen) {
+                setDockedComponentId(undefined);
+              } else {
+                setDockedComponentId(id);
+              }
+            }}
+          />
+        );
+      })}
+    </Menu>
+  );
+  return (
+    <Dropdown overlay={MenuItems} onVisibleChange={setIsMenuOpen} placement="bottom-end">
+      <ToolbarButton icon="web-section" isOpen={isMenuOpen} variant={isOpen ? 'active' : 'default'} />
+    </Dropdown>
+  );
+}
```

Comment:
It would be nice to add some tests for the logic in  `ExtensionToolbarItem` and the `ExtensionSidebarProvider`.